homeassistant/components/anglian_water/__init__.py

```diff
@@ -0,0 +1,70 @@
+"""The Anglian Water integration."""
+
+from __future__ import annotations
+
+from pyanglianwater import AnglianWater
+from pyanglianwater.auth import MSOB2CAuth
+from pyanglianwater.exceptions import (
+    ExpiredAccessTokenError,
+    SelfAssertedError,
+    SmartMeterUnavailableError,
+)
+
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed, ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_ACCOUNT_NUMBER, DOMAIN
+from .coordinator import AnglianWaterConfigEntry, AnglianWaterUpdateCoordinator
+
+_PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: AnglianWaterConfigEntry
+) -> bool:
+    """Set up Anglian Water from a config entry."""
+    auth = MSOB2CAuth(
+        username=entry.data[CONF_USERNAME],
+        password=entry.data[CONF_PASSWORD],
+        session=async_get_clientsession(hass),
+        refresh_token=entry.data[CONF_ACCESS_TOKEN],
+        account_number=entry.data[CONF_ACCOUNT_NUMBER],
+    )
+    try:
```

Comment:
There's overlap here with the conflig_flow logic. Is it worth trying to extract something to avoid the duplication?

Reply:
I'm not sure there is? The account number is set here:

https://github.com/pantherale0/home-assistant/blob/5cd564e57a5807d038a1d20b5893d8d6f159788a/homeassistant/components/anglian_water/config_flow.py#L71